crates/router/src/routes/disputes/utils.rs

```diff
@@ -0,0 +1,102 @@
+use actix_multipart::{Field, Multipart};
+use actix_web::web::Bytes;
+use common_utils::{errors::CustomResult, ext_traits::StringExt, fp_utils};
+use error_stack::{IntoReport, ResultExt};
+use futures::{StreamExt, TryStreamExt};
+
+use crate::{
+    core::{errors, files::helpers},
+    types::api::{disputes, files},
+    utils::OptionExt,
+};
+
+pub async fn parse_evidence_type(
+    field: &mut Field,
+) -> CustomResult<Option<disputes::EvidenceType>, errors::ApiErrorResponse> {
+    let purpose = helpers::read_string(field).await;
+    match purpose {
+        Some(evidence_type) => Ok(Some(
+            evidence_type
+                .parse_enum("Evidence Type")
+                .change_context(errors::ApiErrorResponse::InternalServerError)
+                .attach_printable("Error parsing evidence type")?,
+        )),
+        _ => Ok(None),
+    }
+}
+
+pub async fn get_attach_evidence_request(
+    mut payload: Multipart,
+) -> CustomResult<disputes::AttachEvidenceRequest, errors::ApiErrorResponse> {
+    let mut option_evidence_type: Option<disputes::EvidenceType> = None;
+    let mut dispute_id: Option<String> = None;
+
+    let mut file_name: Option<String> = None;
+    let mut file_content: Option<Vec<Bytes>> = None;
+
+    while let Ok(Some(mut field)) = payload.try_next().await {
+        let content_disposition = field.content_disposition();
+        let field_name = content_disposition.get_name();
+        // Parse the different parameters expected in the multipart request
+        match field_name {
+            Some("file") => {
+                file_name = content_disposition.get_filename().map(String::from);
+                //Collect the file content and throw error if something fails
+                let mut file_data = Vec::new();
+                let mut stream = field.into_stream();
+                while let Some(chunk) = stream.next().await {
+                    match chunk {
+                        Ok(bytes) => file_data.push(bytes),
+                        Err(err) => Err(errors::ApiErrorResponse::InternalServerError)
+                            .into_report()
+                            .attach_printable(format!("{}{}", "File parsing error: ", err))?,
```

Comment:
```suggestion
                            .attach_printable_lazy(|| format!("File parsing error: {err}"))?,
```

`attach_printable_lazy()` is preferred whenever there's a function call (in this case `format!()`) involved.